packages/react-native/React/Base/RCTConvert.m

```diff
@@ -1487,6 +1487,105 @@ + (NSPropertyList)NSPropertyList:(id)json
+      @"feed": NSAccessibilityGroupRole,
+      @"figure": NSAccessibilityGroupRole,
+      @"form": NSAccessibilityGroupRole,
+      @"grid": NSAccessibilityTableRole,
+      @"gridcell": NSAccessibilityCellRole,
+      @"group": NSAccessibilityGroupRole,
+      @"heading": NSAccessibilityStaticTextRole,
+      @"image": NSAccessibilityImageRole,
+      @"img": NSAccessibilityImageRole,
+      @"link": NSAccessibilityLinkRole,
+      @"list": NSAccessibilityListRole,
+      @"listbox": NSAccessibilityListRole,
+      @"listitem": NSAccessibilityGroupRole,
+      @"log": NSAccessibilityGroupRole,
+      @"main": NSAccessibilityGroupRole,
+      @"marquee": NSAccessibilityGroupRole,
+      @"math": NSAccessibilityGroupRole,
+      @"menu": NSAccessibilityMenuRole,
+      @"menubar": NSAccessibilityMenuBarRole,
+      @"menuitem": NSAccessibilityMenuItemRole,
+      @"menuitemcheckbox": NSAccessibilityMenuItemRole,
+      @"menuitemradio": NSAccessibilityMenuItemRole,
+      @"meter": NSAccessibilityLevelIndicatorRole,
+      @"navigation": NSAccessibilityGroupRole,
+      @"none": NSAccessibilityGroupRole,
+      @"note": NSAccessibilityGroupRole,
+      @"option": NSAccessibilityStaticTextRole,
+      @"presentation": NSAccessibilityGroupRole,
+      @"progressbar": NSAccessibilityProgressIndicatorRole,
+      @"radio": NSAccessibilityRadioButtonRole,
+      @"radiogroup": NSAccessibilityRadioGroupRole,
+      @"region": NSAccessibilityGroupRole,
+      @"row": NSAccessibilityRowRole,
+      // rowgroup not mapped
+      @"rowheader": NSAccessibilityCellRole,
+      @"scrollbar": NSAccessibilityScrollBarRole,
+      @"search": NSAccessibilityGroupRole,
+      @"searchbox": NSAccessibilityTextFieldRole,
+      @"separator": NSAccessibilitySplitterRole,
+      @"slider": NSAccessibilitySliderRole,
+      @"spinbutton": NSAccessibilityIncrementorRole,
+      @"status": NSAccessibilityGroupRole,
+      @"switch": NSAccessibilityCheckBoxRole,
+      @"tab": NSAccessibilityRadioButtonRole,
+      @"table": NSAccessibilityTableRole,
+      @"tablist": NSAccessibilityTabGroupRole,
+      @"tabpanel": NSAccessibilityGroupRole,
+      @"term": NSAccessibilityGroupRole,
+      @"textbox": NSAccessibilityTextFieldRole,
+      @"timer": NSAccessibilityGroupRole,
+      @"toolbar": NSAccessibilityToolbarRole,
+      @"tooltip": NSAccessibilityGroupRole,
+      @"tree": NSAccessibilityOutlineRole,
+      @"treegrid": NSAccessibilityTableRole,
+      @"treeitem": NSAccessibilityRowRole,
+    };
+  });
+  NSAccessibilityRole nsRole = [ariaRoleToNsRole valueForKey: ariaRole];
+  if (nsRole == nil) {
+    // Dictionary above is fully filled for roles spec'd by aria mappings since
+    // the aria mappings disagrees with some of the old mappings. We'll still 
+    // fall back to the legacy mappings for macOS specific roles like 
+    // `disclosure` and legacy accessibilityTrait-based mappings like 
+    // `adjustable`.
```

Comment:
Maybe something more succinct like "Fall back to legacy mappings if an aria mapping is not found. This would include macOS specific roles like `disclosure` and legacy accessibilityTrait-based mappings like `adjustable`.". Also for comments longer than 3 lines, I think I prefer `/* .... */